lib/Dialect/TritonGPU/Transforms/WSCodePartition.cpp

```diff
@@ -137,16 +163,45 @@ Operation *SpecializeIfOp(scf::IfOp ifOp, IRMapping &mapping,
     SpecializeOp(&thenOp, mapping, ifBuilder, asyncTaskId);
   }
 
+  // Update yields
+  auto loc = ifOp.getLoc();
+  auto updateYield = [&](scf::YieldOp yield, SmallVector<Value> &operands) {
+    ifBuilder.setInsertionPoint(yield);
+    ifBuilder.createWithAsyncTaskIds<scf::YieldOp>(loc, operands);
+    yield.erase();
+  };
+  SmallVector<Value> ifYieldOperands;
+  resultIdx = 0;
+  for (Value yieldV : newIfOp.thenYield().getOperands()) {
```

Comment:
nit: If we compute `Results` instead of `ignoredResults`, we can iterate over `Results` and call `newIfOp.thenYield().getOperand[i]` instead of iterating over all operands.